perfkitbenchmarker/providers/aws/aws_network.py

```diff
@@ -64,6 +64,27 @@ def __init__(self):
     self.firewall_set = set()
     self._lock = threading.Lock()
 
+  def AllowIcmp(self, vm):
+    """Opens the ICMP protocol on the firewall.
+    Args:
+      vm: The BaseVirtualMachine object to open the ICMP protocol for.
+    """
+    entry = (-1, -1, vm.region, vm.group_id)
```

Comment:
Other `firewall_set` entries include cidr as the fifth tuple.  Should we have different firewall_set for tcp/udp and icmp?

Reply:
Changed to add cidr as a fifth tuple. I don't think we need a separate firewall set for ICMP, they are all firewall rules and with the way the entries are formatted, there shouldn't be duplicates

Reply:
They both (int, int, string, string, string) Seems like there could be duplicates.  Please either use separate sets or add the protocol to the tuple, or add tests to prevent future maintainers from breaking the implicit formatting assumptions.

Reply:
I see where the overlap could be now. I've moved icmp firewall rules to their own sets for both aws and gcp